scrapers/errata.py

```diff
@@ -0,0 +1,188 @@
+        """
+        Update Neo4j with Errata Tool advisories from TEIID
+        :param advisories: a list of dictionaries of advisories
+        :return: None
+        """
+        count = 0
+        for advisory in advisories:
+            count += 1
+            log.info('Processing advisory {0}/{1}'.format(count, len(advisories)))
+            # The content_types column is a string with YAML in it, so convert it to a list
+            content_types = yaml.safe_load(advisories[0]['content_types'])
+            adv = Advisory.get_or_create({
+                'actual_ship_date': advisory['actual_ship_date'],
+                'advisory_name': advisory['advisory_name'],
+                'content_types': content_types,
+                'created_at': advisory['created_at'],
+                'id_': advisory['id'],
+                'issue_date': advisory['issue_date'],
+                'product_name': advisory['product_name'],
+                'product_short_name': advisory['product_short_name'],
+                'release_date': advisory['release_date'],
+                'security_impact': advisory['security_impact'],
+                'security_sla': advisory['security_sla'],
+                'state': advisory['state'],
+                'status_time': advisory['status_time'],
+                'synopsis': advisory['synopsis'],
+                'type_': advisory['type'],
+                'update_date': advisory['update_date'],
+                'updated_at': advisory['updated_at']
+            })[0]
+            assigned_to = User.get_or_create({'username': advisory['assigned_to'].split('@')[0]})[0]
+            adv.assigned_to.connect(assigned_to)
+            assigned_to.advisories_assigned.connect(adv)
+            package_owner = User.get_or_create(
+                {'username': advisory['package_owner'].split('@')[0]})[0]
+            adv.package_owners.connect(package_owner)
+            package_owner.advisories_package_owner.connect(adv)
+            reporter = User.get_or_create({'username': advisory['reporter'].split('@')[0]})[0]
+            adv.reporters.connect(reporter)
+            reporter.advisories_reported.connect(adv)
+            reporter.advisories.connect(adv)
+
+            for state in self.get_advisory_states(advisory['id']):
+                adv_state = AdvisoryState.get_or_create({
+                    'name': state['name'],
+                    'created_at': state['created_at'],
+                    'updated_at': state['updated_at']
+                })[0]
+                adv.states.connect(adv_state)
+                adv_state.advisories.connect(adv)
+                state_owner = User.get_or_create({'username': state['username'].split('@')[0]})[0]
+                adv_state.owner.connect(state_owner)
+                state_owner.advisories_state_owner.connect(adv_state)
+
+            for attached_bug in self.get_attached_bugs(advisory['id']):
+                bug = BugzillaBug.get_or_create(attached_bug)[0]
```

Comment:
I was wondering if even this can use create_or_update

Reply:
Since `attached_bug` is just the `id_` so it will never update. It's best to just do `get_or_create` for efficiency and readability.